### ISSUE TITLE
Extract Zarr reader into its own package

### ISSUE THREAD

**Issue by alxmrs** [google-research/neuralgcm#97]
I would love to be able to install NeuralGCM’s Zarr/tfds reader as its own python package. 

https://github.com/google-research/neuralgcm/blob/d32e8b6365bd6096a76b531a14df1a27723d5e2c/neuralgcm/reference_code/reader.py#L378

It would be nice to see how this compares to xbatcher in benchmarks. I’d also like to see how well it would work with tile-serving-backed Xarray datasets, like data via Xee.

**Comment by shoyer** [comment]
Yes, I would love this! This would definitely make sense on its own project, I will see if I can finish up the open source launch :)
